### ISSUE TITLE
Output of llma-2-7b model is repetative. 

### ISSUE THREAD

**Issue by Vithulep** [alipay/PainlessInferenceAcceleration#32]
**1. We can see that in the output of the prompt same sentence is repeating. Why is this happening?**
Prompt: "AI is going to" 
On other prompt also same repetitions are happening.

![Lookahead_their code fp16](https://github.com/user-attachments/assets/a8d4cf9a-6547-4380-b82e-a0714ad56cb4)
![image](https://github.com/user-attachments/assets/2b134da8-774a-4d84-b369-0b901a31e125)

**2. When we directly run Lookahead (not run for lookahead=false earlier), we will get less throughput [51.3 tokens/sec], and in second run we are getting around [179 tokens/sec]. Can you explain why is this happening? Are you storing the result in cache or are you using the same Trie tree again?**

![Their_Lookahead_after first output](https://github.com/user-attachments/assets/5f3489ce-a8e3-4c09-8c33-f51e2ba6b19c)

Model: llama-2-7b fp16 
file run from repo: llama_example.py

**Comment by zheyishine** [comment]
1. It seems the prompt will generate a repetitive response, even without lookahead.
2. Yes, Trie tree lives until python process finished.

**Comment by Vithulep** [comment]
> 1. It seems the prompt will generate a repetitive response, even without lookahead.
> 2. Yes, Trie tree lives until python process finished.

No. Without lookahead responses were not repeating. (It was happening for some cases not all cases). 
2. Then this is very useful in the case when we have created a tree beforehand of tokens (sentences), and these tokens or patterns we are reusing.  Like FAQs and Not in other where previous occurrence is not repetitive.
